### ISSUE TITLE
[ANDROID][Uno.SkiaSharp] ManipulationEvents not implemented

### ISSUE THREAD

**Issue by peternary** [unoplatform/uno#1976]
## Current behavior

OnManipulationDelta is not fired for SkXamlCanvas on Anroid

## Expected behavior

OnManipulationDelta should work as it does on UWP.

## How to reproduce it (as minimally and precisely as possible)

Load the attached project in UWP and observe that manipulation event is fired for both Grid and SkXamlCanvas.  Load Android project and observe that manipulation events only fire for Grid.

[AndroidManipulation.zip](https://github.com/unoplatform/uno/files/3779453/AndroidManipulation.zip)

## Environment

Nuget Package:

Uno.SkiaSharp.Views

Package Version(s):

1.68.0

Affected platform(s):

- [ ] iOS
- [x] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by jeromelaban** [comment]
This issue is related to https://github.com/unoplatform/uno/issues/1940

**Comment by jeromelaban** [comment]
We can repro with `2.0.512-dev.3874`

The issues is related with the hit testing detection, where a type inheriting directly from FrameworkElement uses the default behavior of a transparent control cannot have manipulations set to it.

A possible workaround is to set a `Background` property to something other than transparent. In the case of the `SKXamlCanvas`, which is generally completely overriden by another color, that may be enough.

@peternary Do you need the Skia canvas to be partially transparent ?

**Comment by peternary** [comment]
@jeromelaban No, we do write a background color.  Will set the Background property to a color as a workaround for now.

**Comment by francoistanguay** [comment]
Isnt this fixed?

**Comment by dr1rrb** [comment]
This might have been fixed by https://github.com/unoplatform/uno/pull/7571

**Comment by michael-hawker** [comment]
Ah, I think I just hit this as well, but not entirely sure as I'm not seeing things layout in my Canvas correctly, can follow-up later this week after tomorrow.

**Comment by michael-hawker** [comment]
I honestly don't remember leaving that comment a few days ago, not sure if it was the exact same circumstance...

I know at that time I was validating our CanvasView component in WASM which was both not displaying things in the correct location, and also ignoring the Manipulation Delta events (or maybe they were received but couldn't be displayed because of the first problem).

As seen here at https://toolkitlabs.dev these boxes shouldn't be all on top of one another, and they should be draggable:

![image](https://user-images.githubusercontent.com/24302614/180503709-02ba1e35-9cb4-437e-87b9-ba36472114a5.png)

**Comment by dr1rrb** [comment]
Well @michael-hawker I tried your sample, and it seems to be a binding issue. The `ManipulationDelta` handler is being invoked and in that handler if I replace the `SetBindingExpression` by a simple `SetValue` I'm able to move the elements:

```cs
            //cp.SetBindingExpressionValue(Canvas.LeftProperty, Canvas.GetLeft(cp) + e.Delta.Translation.X);
            //cp.SetBindingExpressionValue(Canvas.TopProperty, Canvas.GetTop(cp) + e.Delta.Translation.Y);
            cp.SetValue(Canvas.LeftProperty, Canvas.GetLeft(cp) + e.Delta.Translation.X);
            cp.SetValue(Canvas.TopProperty, Canvas.GetTop(cp) + e.Delta.Translation.Y);
```

![image](https://user-images.githubusercontent.com/8635919/180868434-9f210507-7d5f-4f05-92b4-fbe654dee6a8.png)

Has you are already doing the same kind of magic to set the original values of the attached `Canvas.<Left|Top>` properties, it's most probably the same issue. Do you already have an issue for the layouting issue we can link to?

**Comment by michael-hawker** [comment]
@dr1rrb interesting, so there might be an issue with the underlying binding code there in Uno WASM (as it works fine in UWP and WinAppSDK). Will have to investigate further. Don't have any other issues open yet.

**Comment by jeromelaban** [comment]
> @dr1rrb interesting, so there might be an issue with the underlying binding code there in Uno WASM (as it works fine in UWP and WinAppSDK). Will have to investigate further. Don't have any other issues open yet.

@michael-hawker I would not be surprised if Uno were having discrepancies in the way property vs. fields are declared, causing [this line](https://github.com/CommunityToolkit/Labs-Windows/blob/ed6bd23523770f34ebf6b41304dd257374fe1f04/labs/CanvasView/src/FrameworkElementExtensions.cs#L32) to fail silently when a binding is declared.

**Comment by michael-hawker** [comment]
@jeromelaban thanks, yeah I suspect there's something going on there. We still need to update to the latest Uno version, so that's still the first step for us to check-out. We do the special redirect here in order to support the binding pass-through for the templating of how this system works for `Canvas`.

**Comment by MartinZikmund** [comment]
Only relevant to native renderer, closing.
